### DISCUSSION TITLE
Renovate not picking up "regexManagers:dockerfileVersions" pattern in Dockerfile

### DISCUSSION THREAD

**Discussion by candrews** [renovatebot/renovate#18137] [Request Help] [answered]
I have [`regexManagers:dockerfileVersions`](https://docs.renovatebot.com/presets-regexManagers/#regexmanagersdockerfileversions) configured in my project: https://github.com/candrews/wine-adoptopenjdk/blob/master/renovate.json

And in my `Dockerfile` I'm trying to use that functionality: https://github.com/candrews/wine-adoptopenjdk/blob/master/Dockerfile#L14
```dockerfile
# renovate: datasource=adoptium-java depName=java
ARG JAVA_VERSION="17.0.4.1+1"
```

But according to the [Dependency Dashboard issue](https://github.com/candrews/wine-adoptopenjdk/issues/14) renovate isn't picking up this dependency. Oddly, it's picking up other dependencies that use this same approach in the same file.

What am I doing wrong?

Thank you very much in advance!

**Comment by Churro** [comment]
In contrast to other dependencies, no `versioning` field is specified. This causes renovate to default to `semver` (see [here](https://docs.renovatebot.com/configuration-options/#regexmanagers)). As `17.0.4.1+1` doesn't comply with semantic versioning 2.0.0, it is not picked up.

**Reply by candrews** [reply]
Looking at the list of versionings at https://docs.renovatebot.com/modules/versioning/ I'm not sure which would work for Java versions, which are rather odd looking... do you have any suggestions?

Also, Renovate itself has the this problem at https://github.com/renovatebot/docker-java/blob/main/Dockerfile#L1 - I'll submit an MR once I can figure out what the fix is

**Reply by candrews** [reply]
I tried the following and according to the Renovate Dashboard renovate was still unable to discover this dependency:
```dockerfile
# renovate: datasource=adoptium-java depName=java versioning=regex:^(?<major>\d+)?(\.(?<minor>\d+))?(\.(?<patch>\d+))?([\._+](?<build>\d+))?(-(?<compatibility>.*))?$
ARG JAVA_VERSION="17.0.4.1+1"
```

**Comment by Churro** [comment]
`17.0.4.1+1` should be compliant at least with `semver-coerced` and `loose` versioning.

The version in https://github.com/renovatebot/docker-java/blob/main/Dockerfile#L1 has one digit less, making it semver-compliant (see [here](https://regex101.com/r/YbzBdy/1)).

**Reply by candrews** [reply]
That makes sense, thanks! So I've tried `loose` but it still isn't discovering dependency :(

```Dockerfile
# renovate: datasource=adoptium-java depName=java versioning=loose
ARG JAVA_VERSION="17.0.4.1+1"
```

**Reply by rarkins** [reply]
Can you isolate this to a reproduction repo with a single dependency in the Dockerfile, so it's easier to debug?

**Reply by candrews** [reply]
I think this repository should minimally reproduce the issue:
https://github.com/candrews/renovate-dockerfileVersions-test

**Reply by rarkins** [reply]
Hi, can you also do the following?

1. Reduce the Dockerfile size to only minimum required
2. Install the hosted app to demonstrate the problem
3. Clarify in readme.md what the problem is

**Comment by Churro** [comment, marked as answer]
Dependency extraction works correctly using `loose` versioning, see https://github.com/renovate-demo/18137-renovate-dockerfileVersions-test/issues/2

```json
"regex": [
   {
     "packageFile": "Dockerfile",
     "deps": [
       {
         "depName": "alpine_3_13/freetype",
         "currentValue": "2.10.4-r3",
         "datasource": "repology",
         "versioning": "loose",
         "replaceString": "# renovate: datasource=repology depName=alpine_3_13/freetype versioning=loose\nARG FREETYPE_VERSION=\"2.10.4-r3\"\n",
         "depIndex": 0,
         "updates": [],
         "warnings": [],
         "currentVersion": "2.10.4-r3",
         "fixedVersion": "2.10.4-r3"
       },
       {
         "depName": "alpine_3_13/ncurses",
         "currentValue": "6.2_p20210109-r1",
         "datasource": "repology",
         "versioning": "loose",
         "replaceString": "# renovate: datasource=repology depName=alpine_3_13/ncurses versioning=loose\nARG NCURSES_VERSION=\"6.2_p20210109-r1\"\n",
         "depIndex": 1,
         "updates": [],
         "warnings": [],
         "currentVersion": "6.2_p20210109-r1",
         "fixedVersion": "6.2_p20210109-r1"
       },
       {
         "depName": "alpine_3_13/wine",
         "currentValue": "4.0.3-r0",
         "datasource": "repology",
         "versioning": "loose",
         "replaceString": "# renovate: datasource=repology depName=alpine_3_13/wine versioning=loose\nARG WINE_VERSION=\"4.0.3-r0\"\n",
         "depIndex": 2,
         "updates": [],
         "warnings": [],
         "currentVersion": "4.0.3-r0",
         "fixedVersion": "4.0.3-r0"
       },
       {
         "depName": "java",
         "currentValue": "17.0.4.1-1",
         "datasource": "adoptium-java",
         "versioning": "regex:^(?<major>\\d+)?(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?([\\._+](?<build>\\d+))?(-(?<compatibility>.*))?$",
         "replaceString": "# renovate: datasource=adoptium-java depName=java versioning=loose\nARG JAVA_VERSION=\"17.0.4.1-1\"\n",
         "depIndex": 3,
         "updates": [],
         "warnings": [],
         "homepage": "https://adoptium.net",
         "currentVersion": "17.0.4.1-1",
         "fixedVersion": "17.0.4.1-1"
       }
     ],
     "matchStrings": [
       "# renovate: datasource=(?<datasource>[a-z-]+?) depName=(?<depName>[^\\s]+?)(?: (lookupName|packageName)=(?<packageName>[^\\s]+?))?(?: versioning=(?<versioning>[a-z-0-9]+?))?\\s(?:ENV|ARG) .+?_VERSION[ =]\"?(?<currentValue>.+?)\"?\\s"
     ]
   }
]
```
